hilti/toolchain/src/compiler/visitors/operator-resolver.cc

```diff
@@ -152,6 +153,17 @@ struct Normalizer : public hilti::visitor::PostOrder<void, Normalizer> {
             }
         }
     }
+
+    void operator()(const expression::Assign& assign, position_t p) {
+        auto& lhs = assign.childs().front();
+        if ( const auto& index_non_const = lhs.tryAs<operator_::map::IndexNonConst>() ) {
+            const auto map = index_non_const->op0();
+            const auto key = index_non_const->op1();
+
+            auto index_assign = builder::memberCall(std::move(map), "index_assign", {std::move(key), assign.source()});
```

Comment:
I think I'd prefer to transform this into a new operator `IndexAssign` that's overloaded by `map`, instead of a `map` method. That would be more natural to extend to other types needing a custom index-assign operation (e.g., vector). But this is also related to the question if need the explicit method `index assign`, see below.